rskj-core/src/main/java/co/rsk/db/importer/provider/index/BootstrapIndexMCH.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * This file is part of RskJ
+ * Copyright (C) 2019 RSK Labs Ltd.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package co.rsk.db.importer.provider.index;
+
+import co.rsk.db.importer.BootstrapImportException;
+import co.rsk.db.importer.provider.index.data.BootstrapDataEntry;
+import co.rsk.db.importer.provider.index.data.BootstrapDataIndex;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class BootstrapIndexMCH {
+
+    private final List<String> publicKeys;
+    private final int minimumAcceptableSources;
+
+    public BootstrapIndexMCH(List<String> publicKeys, int minimumAcceptableSources) {
+        this.publicKeys = new ArrayList<>(publicKeys);
+        this.minimumAcceptableSources = minimumAcceptableSources;
+    }
+
+    public MaximumCommonHeightData getMaximumCommonHeightData(List<BootstrapDataIndex> indexes) {
```

Comment:
I'd add some documentation as it took me some time to understand the state. I find the Map of Maps hard to follow.

Maybe this alternative implementation?
```
//Group by height
Map<Long, Long> entriesPerHeight = entries.stream().collect(
        Collectors.groupingBy(
                e -> e.height, Collectors.counting())
        );

//Maximum common height
int minimumAcceptableNumber = 2;
long maximumCommonHeight = entriesPerHeight.entrySet().stream()
        .filter(e -> e.getValue() >= minimumAcceptableNumber)
        .map(Map.Entry::getKey).max(Long::compare).orElseThrow(
        () -> new IllegalStateException("Downloaded files doesn't contain enough entries for a common height")
);

//Getting the maximum common height entries
Set<BootstrapDataEntry> maximumCommonEntries =
        entries.stream().filter(e -> e.height == maximumCommonHeight)
                .collect(Collectors.toSet());
```

Reply:
I like the idea about collecting by height but it's important to validate that there aren't more than one entry for the same height and the same source (public key). I'll add some documentation to explain how data is organized.